runhouse/resources/functions/aws_lambda.py

```diff
@@ -0,0 +1,887 @@
+
+    #
+    def starmap(self, args_lists, **kwargs):
+        """Like :func:`map` except that the elements of the iterable are expected to be iterables
+        that are unpacked as arguments. An iterable of [(1,2), (3, 4)] results in [func(1,2), func(3,4)].
+
+        Example:
+            >>> arg_list = [(1,2), (3, 4)]
+            >>> # runs the function twice, once with args (1, 2) and once with args (3, 4)
+            >>> output = my_lambda.starmap(arg_list) # output = ["3", "7"]
+        """
+
+        return [self._invoke(*args, **kwargs) for args in args_lists]
+
+    # --------------------------------------
+    # Properties setup
+    # --------------------------------------
+    @property
+    def config_for_rns(self):
+        config = super().config_for_rns
+        config.update(
+            {
+                "paths_to_code": self.local_path_to_code,
+                "handler_function_name": self.handler_function_name,
+                "runtime": self.runtime,
+                "timeout": self.timeout,
+                "memory_size": self.memory_size,
+                "tmp_size": self.tmp_size,
+                "args_names": self.args_names,
+                "function_arn": self.aws_lambda_config["Configuration"]["FunctionArn"],
+            }
+        )
+        return config
+
+    @property
+    def handler_path(self):
+        return self.local_path_to_code[0]
+
+    # --------------------------------------
+    # Lambda Function delete methods
+    # --------------------------------------
+    def delete(self):
+        try:
+            lambda_name = self.name
+            # delete from rns (if exists)
+            if rns_client.exists(self.rns_address):
+                rns_client.delete_configs(self)
+            # delete from aws console
+            if self._lambda_exist(lambda_name):
+                policy_name = f"{lambda_name}_Policy"
+                role_name = f"{lambda_name}_Role"
+                log_group_name = f"/aws/lambda/{lambda_name}"
+
+                # TODO: do we want to delete role and log group?
+                #  maybe we want to reuse the role or save logs even if the lambda was deleted.
+                iam_client = boto3.client("iam")
+                iam_client.delete_role_policy(
+                    RoleName=role_name, PolicyName=policy_name
+                )
+
+                iam_client.delete_role(RoleName=role_name)
+
+                if len(self._get_log_group_names(log_group_name)) > 0:
+                    self.logs_client.delete_log_group(logGroupName=log_group_name)
+
+                self.lambda_client.delete_function(FunctionName=lambda_name)
+
+            return True
+        except botocore.exceptions.ClientError as aws_exception:
+            raise RuntimeError(
+                f"Could nor delete an AWS resource, got {aws_exception.response['Error']['Message']}"
+            )
+
+
+def _paths_to_code_from_fn_pointers(fn_pointers):
```

Comment:
small nit - can this either be moved to a utils file or the lambdas class? feels a little out of place here with the factory and parent class